HPXMLtoOpenStudio/measure.rb

```diff
@@ -4169,15 +4191,102 @@ def self.set_surface_exterior(model, spaces, surface, exterior_adjacent_to)
+
+    # return if already exists
+    return if not @mf_temp_sch_map[outside_space].nil?
+
+    @mf_temp_sch_map[outside_space] = OpenStudio::Model::ScheduleConstant.new(model)
+    @mf_temp_sch_map[outside_space].setName("#{outside_space}")
+
+    if outside_space == HPXML::LocationOtherHeatedSpace
+      # Average of indoor/outdoor temperatures with minimum of 68 deg-F
+      temp_min = UnitConversions.convert(68, 'F', 'C')
+      indoor_weight = 0.5
+      outdoor_weight = 0.5
+    elsif outside_space == HPXML::LocationOtherMultifamilyBufferSpace
+      # Average of indoor/outdoor temperatures with minimum of 50 deg-F
+      temp_min = UnitConversions.convert(50, 'F', 'C')
+      indoor_weight = 0.5
+      outdoor_weight = 0.5
+    elsif outside_space == HPXML::LocationOtherNonFreezingSpace
+      # Floating with outdoor air temperature with minimum of 40 deg-F
+      temp_min = UnitConversions.convert(40, 'F', 'C')
+      indoor_weight = 0.0
+      outdoor_weight = 1.0
+    elsif outside_space == HPXML::LocationOtherHousingUnit
+      # For water heater, duct, appliances, etc.
+      # Indoor air temperature
+      temp_min = UnitConversions.convert(40, 'F', 'C')
+      indoor_weight = 1.0
+      outdoor_weight = 0.0
+    end
+
+    # Schedule type limits compatible
+    schedule_type_limits = OpenStudio::Model::ScheduleTypeLimits.new(model)
+    schedule_type_limits.setUnitType('Temperature')
+    @mf_temp_sch_map[outside_space].setScheduleTypeLimits(schedule_type_limits)
+
+    # Ems to actuate schedule
+    sensor_ia = OpenStudio::Model::EnergyManagementSystemSensor.new(model, 'Zone Air Temperature')
+    sensor_ia.setName('cond_zone_temp')
+    sensor_ia.setKeyName(create_or_get_space(model, spaces, 'living space').name.to_s)
+
+    sensor_oa = OpenStudio::Model::EnergyManagementSystemSensor.new(model, 'Site Outdoor Air Drybulb Temperature')
+    sensor_oa.setName('oa_temp')
+
+    actuator = OpenStudio::Model::EnergyManagementSystemActuator.new(@mf_temp_sch_map[outside_space], 'Schedule:Constant', 'Schedule Value')
+    actuator.setName("#{outside_space.gsub(' ', '_').gsub('-', '_')}_temp_sch")
+
+    program = OpenStudio::Model::EnergyManagementSystemProgram.new(model)
+    program.setName('Other Side Indoor Temperature Program')
+    program.addLine("Set #{actuator.name} = #{sensor_ia.name} * #{indoor_weight} + #{sensor_oa.name} * #{outdoor_weight}")
+    program.addLine("If #{actuator.name} < #{temp_min}")
+    program.addLine("Set #{actuator.name} = #{temp_min}")
+    program.addLine('EndIf')
+
+    program_cm = OpenStudio::Model::EnergyManagementSystemProgramCallingManager.new(model)
+    program_cm.setName("#{program.name} calling manager")
+    program_cm.setCallingPoint('EndOfSystemTimestepAfterHVACReporting')
+    program_cm.addProgram(program)
+  end
+
   # Returns an OS:Space, or nil if the location is outside the building
   def self.get_space_from_location(location, object_name, model, spaces)
```

Comment:
This is very confusing that the method says it returns a space, the objects assigned (e.g., `duct_space`) suggest it's a space, but it may be a schedule now. Let's update the name of this method to `get_space_or_temp_schedule_from_location` and update the comment above too. Then we should return either` [space, nil]` or `[nil, schedule]` so we can do, e.g.,

duct_space, duct_temp_schedule = get_space_or_temp_schedule_from_location(...)

Then subsequent code can use one or the other appropriately.

Reply:
Good idea, I admit that this part is pretty confusing.